### PR TITLE
Bugfix 25831

### PR DESCRIPTION
**Pull request by playamongthestars**
Mantis Issue: https://mantis.ilias.de/view.php?id=25831

### CODE REVIEW

File: Modules/Forum/classes/class.ilForumPostsDeleted.php [hunk 1]
```diff
@@ -89,7 +89,16 @@ public function __construct(ilObjForumNotificationDataProvider $provider = NULL)
 		
 		if(is_object($provider))
 		{
-			$this->setDeletedBy($this->user->getLogin());
+			if($provider->objPost->getUserAlias() && $provider->objPost->getDisplayUserId() == 0
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilForumPostsDeleted.php]: Can you please extract this condition to a method with a meaningful name?

File: Modules/Forum/classes/class.ilObjForumNotificationDataProvider.php [hunk 2]
```diff
@@ -486,6 +486,15 @@ public function getPosAuthorId()
 	public function getDeletedBy()
 	{
 		global $DIC;
-		return $DIC->user()->getLogin();
+
+		if($this->objPost->getUserAlias() && $this->objPost->getDisplayUserId() == 0
```
Comment (mjansenDatabay) [inline Modules/Forum/classes/class.ilObjForumNotificationDataProvider.php]: Can you please extract this condition to a method with a meaningful name?

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]
LGTM, please use a `Extract Method` refactoring as mentioned in my inline comment.

File: Modules/Forum/classes/class.ilObjForumNotificationDataProvider.php (hunk 2 above)
Comment (playamongthestars) [reply Modules/Forum/classes/class.ilObjForumNotificationDataProvider.php]: done

File: Modules/Forum/classes/class.ilForumPostsDeleted.php (hunk 1 above)
Comment (playamongthestars) [reply Modules/Forum/classes/class.ilForumPostsDeleted.php]: done

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**